department-of-veterans-affairs/caseflow
# Added time slot component behind feature flag

Resolves [CASEFLOW-955](https://vajira.max.gov/browse/CASEFLOW-955)

### Description
Adds 2 new components `TimeSlot` and `TimeSlotButton` and hides the new components behind a feature flag. Additionally, there is placeholder calls to the backend for when we connect this PR

### Acceptance Criteria
- [x] put work behind feature toggle
- [x] one storybook component for a single button which will display as design (time, issue count, appeal type, poa name, disable flag)
- [x] another storybook component that displays list of above buttons given a list of data
  - list of data would include available slots + taken slots 
  - available slots would be generated based on the 8:30 RO time start time to 3:30 est end time
- [x] taken slots should be disabled
  - use fake data (collaborate with the person working https://vajira.max.gov/browse/CASEFLOW-954 on data format) to generate button components on the fly
- [x] When a hearing is scheduled for a custom time that does not match a time slot, hide any time slot buttons that are less than one hour after the custom time (e.g., after a 12:15pm custom time the 12:30pm button would be hidden but 1:30pm and all buttons after would be visible)
- [x] allow the toggle of old style dropdown
- [x] Time permitting: do the Redux work described in CASEFLOW-956

### Testing Plan
1. Shadow user `BVASYELLOW`: http://localhost:3000/test/users
2. Navigate to the hearings app and click "Schedule Veterans"
3. Search for "St. Petersburg FL" in the list and choose one of the veterans in the AMA waiting tab
4. Click the "Schedule Veteran" button from the dropdown
5. Ensure that you are still able to schedule a veteran with the existing radio buttons
6. On the rails console, enable the feature flag: `FeatureToggle.enable!(:enable_hearing_time_slots)`
7. Repeat steps 2 - 5
8. Ensure that this time you see the new time slot options

### User Facing Changes

 BEFORE|AFTER
 ---|---
-- | <img width="997" alt="Screen Shot 2021-02-25 at 9 01 02 AM" src="https://user-images.githubusercontent.com/61989526/109189724-2dd23180-7749-11eb-913b-47a251944ea3.png">
-- |<img width="1031" alt="Screen Shot 2021-02-25 at 9 01 13 AM" src="https://user-images.githubusercontent.com/61989526/109189751-36c30300-7749-11eb-807d-8f8caa23dee1.png">

### Code Documentation Updates

### Storybook Story
*For Frontend (Presentationa) Components*
